src/core/providers/ogr/qgsgeopackageproviderconnection.cpp

```diff
@@ -336,6 +336,10 @@ void QgsGeoPackageProviderConnection::setDefaultCapabilities()
   mCapabilities |= Capability::ListFieldDomains;
 #endif
 
+#if GDAL_VERSION_NUM >= GDAL_COMPUTE_VERSION(3,6,0)
```

Comment:
This might well kick me into building gdal locally instead of relying on the distro package :)